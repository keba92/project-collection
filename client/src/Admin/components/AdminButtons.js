import React, { memo } from 'react';
import { useTranslation } from 'react-i18next';

function AdminButtons(props) {
    const { blockUser, deleteUsers, makeAdmin } = props;
    const { t, i18n } = useTranslation();

    return (
        <span className = 'toolBar'>
            <button className = 'btn' ><i className='fa fa-close' title='block' onClick={blockUser}>{t('blockI')}</i></button>
            <button className = 'btn' ><i className='fa fa-trash' onClick={deleteUsers}>{t('deleteI')}</i></button>
            <button className = 'btn' ><i className='fa fa-check' title='unblock' onClick={blockUser}>{t('unblockI')}</i></button>
            <button className = 'btn' onClick={makeAdmin}><i className='fa fa-check'>{t('adminI')}</i></button>
        </span>
    )
}

export default memo(AdminButtons);