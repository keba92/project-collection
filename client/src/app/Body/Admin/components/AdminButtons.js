import React, { memo } from 'react';
import { useTranslation } from 'react-i18next';
import PropTypes from 'prop-types';

function AdminButtons(props) {
  const { blockUser, deleteUsers, makeAdmin } = props;
  const { t, i18n } = useTranslation();

  return (
    <span className="toolBar">
      <button className="btn">
        <i className="fa fa-close" title="block" onClick={blockUser}>
          {t('blockI')}
        </i>
      </button>
      <button className="btn">
        <i className="fa fa-trash" onClick={deleteUsers}>
          {t('deleteI')}
        </i>
      </button>
      <button className="btn">
        <i className="fa fa-check" title="unblock" onClick={blockUser}>
          {t('unblockI')}
        </i>
      </button>
      <button className="btn" onClick={makeAdmin}>
        <i className="fa fa-check">{t('adminI')}</i>
      </button>
    </span>
  );
}

AdminButtons.propTypes = {
  blockUser: PropTypes.func,
  deleteUsers: PropTypes.func,
  makeAdmin: PropTypes.func,
};

export default memo(AdminButtons);
