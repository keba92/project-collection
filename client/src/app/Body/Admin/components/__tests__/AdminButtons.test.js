import React from 'react';
import AdminButtons from '../AdminButtons';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

afterEach(cleanup);

const testFn = ()=>{
    return 'test';
}

it('matches snapshot', () => {
    const tree = renderer.create(<AdminButtons blockUser={testFn} deleteUsers={testFn} makeAdmin={testFn} />).toJSON();
    expect(tree).toMatchSnapshot();
})
