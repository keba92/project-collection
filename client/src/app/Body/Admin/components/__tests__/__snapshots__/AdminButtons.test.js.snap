// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<span
  className="toolBar"
>
  <button
    className="btn"
  >
    <i
      className="fa fa-close"
      onClick={[Function]}
      title="block"
    >
      blockI
    </i>
  </button>
  <button
    className="btn"
  >
    <i
      className="fa fa-trash"
      onClick={[Function]}
    >
      deleteI
    </i>
  </button>
  <button
    className="btn"
  >
    <i
      className="fa fa-check"
      onClick={[Function]}
      title="unblock"
    >
      unblockI
    </i>
  </button>
  <button
    className="btn"
    onClick={[Function]}
  >
    <i
      className="fa fa-check"
    >
      adminI
    </i>
  </button>
</span>
`;
